sdk/typescript/src/types/coin.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { array, Infer, literal, nullable, number, object, optional, string, union } from 'superstruct';
+import { ObjectId, TransactionDigest } from './common';
+
+export const CoinStruct = object({
+    coinType: string(),
+    coinObjectId: ObjectId,
+    version: number(),
+    digest: TransactionDigest,
+    balance: number(),
+    lockedUntilEpoch: nullable(number()),
+});
+
+  
+export type CoinStruct = Infer<typeof CoinStruct>;
+  
+export const PaginatedCoins = object({
+    data: array(CoinStruct),
+    nextCursor: union([ObjectId, literal(null)]),
+});
+  
+export type PaginatedCoins = Infer<typeof PaginatedCoins>;
+
+export const BalanceStruct = object({
+    coinType: string(),
+    coinObjectCount: number(),
+    totalBalance: number(),
+    lockedBalance: object({epochId: optional(number()), number: optional(number())})
+});
+
+export type BalanceStruct = Infer<typeof BalanceStruct>;
+
+export const SupplyStruct = object({
+    value: number()
+});

```

Comment:
I would call these `CoinBalance` and `CoinSupply` instead of `*struct` since struct does not convey any meaning here.